docs/src/content/docs/components/file-tree.mdx

```diff
@@ -0,0 +1,237 @@
+
+import Preview from '~/components/component-preview.astro';
+
+<Preview>
+
+<FileTree slot="preview">
+
+- astro.config.mjs an **important** file
+- package.json
+- README.md
+- src
+  - components
+    - **Header.astro**
+  - …
+- pages/
+
+</FileTree>
+
+</Preview>
+
+## Import
+
+```tsx
+import { FileTree } from '@astrojs/starlight/components';
+```
+
+## Usage
+
+Display a file tree with file icons and collapsible sub-directories using the `<FileTree>` component.
+
+Specify the structure of your files and directories with an [unordered Markdown list](https://www.markdownguide.org/basic-syntax/#unordered-lists) inside `<FileTree>`.
+Create a sub-directory using a nested list or add a `/` to the end of a list item to render it as a directory without specific content.
+
+<Preview>
+
+```mdx
+# src/content/docs/example.mdx
+
+import { FileTree } from '@astrojs/starlight/components';
+
+<FileTree>
+
+- astro.config.mjs
+- package.json
+- src
+  - components
+    - Header.astro
+    - Title.astro
+
+</FileTree>
+```
+
+<Fragment slot="markdoc">
+
+```markdoc
+# src/content/docs/example.mdoc
+
+{% filetree %}
+- astro.config.mjs
+- package.json
+- src
+  - components
+    - Header.astro
+    - Title.astro
+{% /filetree %}
+```
+
+</Fragment>
+
+<FileTree slot="preview">
+
+- astro.config.mjs
+- package.json
+- src
+  - components
+    - Header.astro
+    - Title.astro
+
+</FileTree>
+
+</Preview>
+
+### Highlight entries
+
+Make a file or directory stand out by making its name bold, e.g. `**README.md**`.
+
+<Preview>
+
+```mdx {9}
+# src/content/docs/example.mdx
+
+import { FileTree } from '@astrojs/starlight/components';
+
+<FileTree>
+
+- src
+  - components
+    - **Header.astro**
+    - Title.astro
+
+</FileTree>
+```
+
+<Fragment slot="markdoc">
+
+```markdoc {6}
+# src/content/docs/example.mdoc
+
+{% filetree %}
+- src
+  - components
+    - **Header.astro**
+    - Title.astro
+{% /filetree %}
+```
+
+</Fragment>
+
+<FileTree slot="preview">
+
+- src
+  - components
+    - **Header.astro**
+    - Title.astro
+
+</FileTree>
+
+</Preview>
+
+### Add comments
+
+Add a comment to a file or directory by adding more text after the name.
+The comment can be any text, including Markdown formatting.
+
+<Preview>
+
+```mdx {9}
+# src/content/docs/example.mdx
+
+import { FileTree } from '@astrojs/starlight/components';
+
+<FileTree>
+
+- src
+  - components
+    - Header.astro an **important** file
+    - Title.astro
+
+</FileTree>
+```
+
+<Fragment slot="markdoc">
+
+```markdoc {6}
+# src/content/docs/example.mdoc
+
+{% filetree %}
+- src
+  - components
+    - Header.astro an **important** file
+    - Title.astro
+{% /filetree %}
+```
+
+</Fragment>
+
+<FileTree slot="preview">
+
+- src
+  - components
+    - Header.astro an **important** file
+    - Title.astro
+
+</FileTree>
+
+</Preview>
+
+### Add placeholders
+
+Add placeholder files and directories by using either `...` or `…` as the name.
```

Comment:
This feels like it might benefit from a small explanation for _why_ you might want to show a placeholder:

```suggestion
Add placeholder files and directories by using either `...` or `…` as the name.
This can be useful to indicate to a reader that a folder is expected to contain more items without specifying them all explicitly.
```